diff --git a/__analysis__.py b/__analysis__.py
index 490a0b5..b2eff4c 100644
--- a/__analysis__.py
+++ b/__analysis__.py
@@ -236,7 +236,7 @@ class analysis(object) :
             assert not nonSteps, "\n\nWarning, the following items from listOfSteps() are not analysisSteps:\n"+('\n'.join(' '+str(s) for s in nonSteps))
             for step in filter(lambda s: s.only not in ['','data' if tup.lumi else 'sim'], adjustedSteps) : step.disabled = True
 
-            return analysisLooper(mainTree=self.mainTree(),
+            return analysisLooper(mainTree=self.mainTree(conf["tag"]),
                                   otherTreesToKeepWhenSkimming=self.otherTreesToKeepWhenSkimming(),
                                   nEventsMax=nEventsMax,
                                   leavesToBlackList=self.leavesToBlackList(),
diff --git a/__wrappedChain__.py b/__wrappedChain__.py
index 9ee5094..f3f7a8c 100644
--- a/__wrappedChain__.py
+++ b/__wrappedChain__.py
@@ -63,7 +63,7 @@ class wrappedChain(dict) :
                 continue
 
             for iTreeEntry in range( nTreeEntries )  :
-                if (not nTree) and iTreeEntry==100 : tree.StopCacheLearningPhase()
+                if (not nTree) and tree and iTreeEntry==100 : tree.StopCacheLearningPhase()
                 self.entry = iTreeFirstEntry + iTreeEntry
                 if nEntries!=None and nEntries <= self.entry : self.entry-=1; return
                 self.__localEntry = iTreeEntry
